src/engine/source/router/include/router/policyManager.hpp

```diff
@@ -25,8 +25,8 @@ class PolicyManager
 
 private:
     /* Status */
-    std::unordered_map<std::string, std::vector<RuntimePolicy>> m_policies; ///< Map of policies
-    std::shared_mutex m_mutex;                                              ///< Mutex to protect the policies map
+    std::unordered_map<std::string, std::vector<std::unique_ptr<RuntimePolicy>>> m_policies; ///< Map of policies
```

Comment:
Why do we use an unique_ptr now?

Reply:
Because RuntimePolicy now has class members that are not copyable. m_traceBuffer is a map that contains std::stringstream and is a non-copiable type. To allow copying of the RuntimePolicy, you would have to implement the copy constructor and the copy assignment operator in a custom way. However, this can be tricky and error prone, especially when dealing with smart pointers and complex containers. Instead of trying to make the RuntimePolicy copyable, a safer and easier solution is to use smart pointers.